api/src/controllers/roadmap.ts

```diff
@@ -1,27 +1,76 @@
-import Roadmap from '../models/roadmap';
 import { router, userProcedure } from '../helpers/trpc';
-import { z } from 'zod';
-import { mongoRoadmap, MongoRoadmap } from '@peterportal/types';
+import { SavedPlannerData, savedRoadmap, SavedRoadmap, TransferData } from '@peterportal/types';
+import { db } from '../db';
+import { planner, transferredCourse, user } from '../db/schema';
+import { and, asc, eq, inArray, not } from 'drizzle-orm';
 
 const roadmapsRouter = router({
   /**
    * Get a user's roadmap
    */
-  get: userProcedure.input(z.object({ userID: z.string() })).query(async ({ input }) => {
-    const roadmap = await Roadmap.findOne<MongoRoadmap>({ userID: input.userID });
+  get: userProcedure.query(async ({ ctx }) => {
+    const [planners, transfers, timestamp] = await Promise.all([
+      db
+        .select({ id: planner.id, name: planner.name, content: planner.years })
+        .from(planner)
+        .where(eq(planner.userId, ctx.session.userId!))
+        .orderBy(asc(planner.id)),
+      db
+        .select({ name: transferredCourse.courseName, units: transferredCourse.units })
+        .from(transferredCourse)
+        .where(eq(transferredCourse.userId, ctx.session.userId!)),
+      db.select({ timestamp: user.lastRoadmapEditAt }).from(user).where(eq(user.id, ctx.session.userId!)),
+    ]);
+    const roadmap: SavedRoadmap = {
+      planners: planners as SavedPlannerData[],
+      transfers: transfers as TransferData[],
+      timestamp: timestamp[0].timestamp?.toISOString(),
+    };
     return roadmap;
   }),
   /**
    * Save a user's roadmap
    */
-  save: userProcedure.input(mongoRoadmap).mutation(async ({ input }) => {
-    const { userID, roadmap, coursebag } = input;
-    if (await Roadmap.exists({ userID })) {
-      return await Roadmap.replaceOne({ userID }, { roadmap, userID, coursebag });
-    } else {
-      // add roadmap to mongo
-      return await new Roadmap({ roadmap, userID, coursebag }).save();
-    }
+  save: userProcedure.input(savedRoadmap).mutation(async ({ input, ctx }) => {
+    const { planners, transfers, timestamp } = input;
+    const userId = ctx.session.userId!;
+
+    const plannerUpdates = planners
+      .filter((planner) => planner.id !== undefined)
+      .map((planner_) =>
```

Comment:
if `planner` is an import that you don't want to override, I'd rather have this variable name be `plannerData`